deep_quoridor/src/quoridor_env.py

```diff
@@ -0,0 +1,203 @@
+                    "walls": spaces.Box(0, 1, (self.wall_size, self.wall_size, 2), dtype=np.int8),
+                    "walls_remaining": spaces.Discrete(self.max_walls + 1),
+                    "action_mask": spaces.Box(0, 1, (self.board_size**2 + (self.wall_size**2) * 2,), dtype=np.int8),
+                }
+            )
+            for agent in self.possible_agents
+        }
+
+        self.action_spaces = {
+            agent: spaces.Discrete(self.board_size**2 + (self.wall_size**2) * 2) for agent in self.possible_agents
+        }
+
+        self.reset()
+
+    def reset(self, seed=None, options=None):
+        self.walls = np.zeros((self.wall_size, self.wall_size, 2), dtype=np.int8)
+        self.walls_remaining = {agent: self.max_walls for agent in self.possible_agents}
+        # Positions are (row, col)
+        self.positions = {"player_1": (4, 0), "player_2": (4, 8)}
+
+        self.rewards = {agent: 0 for agent in self.possible_agents}
+        self.terminations = {agent: False for agent in self.possible_agents}
+        self.truncations = {agent: False for agent in self.possible_agents}
+        self.agent_selection = self.agent_order[0]
+
+    def step(self, action):
+        """
+        Players move by selecting an index from 0-80 (9×9 board).
+        Wall placement is mapped to 81-208 (8×8×2).
+        """
+        agent = self.agent_selection
+        if self.terminations[agent]:
+            self._next_player()
+            return
+
+        (row, col, action_type) = self.action_index_to_params(action)
+        if action_type == 0:
+            self._move(agent, (row, col))
+        else:
+            self._place_wall(agent, (row, col), action_type - 1)
+
+        if self._check_win(agent):
+            self.terminations = {a: True for a in self.possible_agents}
+            self.rewards[agent] = 1
+            self.rewards[self._opponent(agent)] = -1
+
+        self._next_player()
+
+    def action_params_to_index(self, row: int, col: int, action_type: int = 0) -> int:
+        """
+        Takes action parameters (row, col, movement_type) to an action index
+        movement_type = 0 for moving, 1 for horizontal wall placement, 2 for vertical wall placement
+        """
+        if action_type == 0:
+            return row * self.board_size + col
+        elif action_type == 1:
+            return self.board_size**2 + row * self.wall_size + col
+        elif action_type == 2:
+            return self.board_size**2 + self.wall_size**2 + row * self.wall_size + col
+        else:
+            raise ValueError(f"Invalid action type: {action_type}")
+
+    def action_index_to_params(self, idx) -> Tuple[int, int, int]:
+        """
+        Takes an action index to action parameters (row, col, movement_type)
+        movement_type = 0 for moving, 1 for horizontal wall placement, 2 for vertical wall placement
+        """
+        action_type, _ = divmod(idx - self.board_size**2, self.wall_size**2)
+        action_type += 1
```

Comment:
This is a very ingenuous way of computing the action type. (BTW, if you're discarding the mod, you can just use integer division with `//`, e.g. `action_type = idx - self.board_size**2 //self.wall_size**2 + 1`
Having said that, I think it would be clearer (and less probabilities of introducing a bug now or later) to go the dumb way and just put `action_type = 1` in the first if branch and so on.
Anyway, feel free to leave it like this if you prefer, just wanted to give you my AR$20 (which is about $0.02)

Reply:
Ok, I made it more boring as you requested. Now our remaining 999 engineers will be able to maintain this more easily :-P